phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -0,0 +1,459 @@
+                    for (String indexName : entry.getValue()) {
+                        if (!dryRun) {
+                            admin.disableTable(TableName.valueOf(indexName));
+                        }
+                        LOGGER.info("Disabled index table "+ indexName);
+                        TableDescriptorBuilder indexTableDescBuilder = TableDescriptorBuilder.newBuilder(admin.getDescriptor(TableName.valueOf(indexName)));
+                        if (admin.getDescriptor(TableName.valueOf(indexName)).hasCoprocessor(GlobalIndexChecker.class.getName())) {
+                            indexTableDescBuilder.removeCoprocessor(GlobalIndexChecker.class.getName());
+                            LOGGER.info("Unloaded GlobalIndexChecker coprocessor on index table "+indexName);
+                        }
+                        if (!dryRun) {
+                            admin.modifyTable(indexTableDescBuilder.build());
+                        }
+                    }
+                    for (String indexName : entry.getValue()) {
+                        if (!dryRun) {
+                            admin.enableTable(TableName.valueOf(indexName));
+                        }
+                        LOGGER.info("Enabled index table "+ indexName);
+                    }
+                    if (!dryRun) {
+                        admin.enableTable(TableName.valueOf(fullTableName));
+                    }
+                    LOGGER.info("Enabled data table "+ fullTableName);
+                } catch (IOException | SQLException e) {
+                    LOGGER.severe("Something went wrong while rolling back the feature "+ e);
+                }
+            }
+
+        } catch (SQLException e) {
+            LOGGER.severe("Something went wrong in rollbackConsistencyFeature "+ e);
+        }
+    }
+
+    private static void checkAndWaitForMajorCompaction(Admin admin, Map.Entry<String, ArrayList<String>> value) {
+        value.getValue().parallelStream().forEach(table -> {
+            try {
+                CompactionState cs = admin.getCompactionState(TableName.valueOf(table));
+                while (cs.equals(CompactionState.MAJOR)) {
+                    LOGGER.fine("Major compaction state is " + cs.name() +" for "+table+" sleeping for 5 secs before next check");
+                    if (!dryRun) {
+                        Thread.sleep(5000);
+                        cs = admin.getCompactionState(TableName.valueOf(table));
+                    }
+                }
+                if (!dryRun) {
+                    LOGGER.fine("Major compaction state is " + cs.name() + " for " + table);
+                }
+            } catch (InterruptedException | IOException e) {
+                LOGGER.severe("Something went wrong while checking and waiting for Major Compaction for table " + table+e);
+            }
+        });
+    }
+
+    private static void resetMajorCompaction(Admin admin, Map.Entry<String, ArrayList<String>> value) {
```

Comment:
This and the below function could be the same with an additional boolean parameter (enable/disable). Call it "setCompactionEnabled" to be in-sync with the admin API.